src/github.com/getlantern/fronted/direct.go

```diff
@@ -29,20 +32,28 @@ var (
 )
 
 type direct struct {
-	tlsConfigsMutex sync.Mutex
-	tlsConfigs      map[string]*tls.Config
-	certPool        *x509.CertPool
-	candidates      chan *Masquerade
-	masquerades     chan *Masquerade
+	tlsConfigsMutex     sync.Mutex
+	tlsConfigs          map[string]*tls.Config
+	certPool            *x509.CertPool
+	candidates          chan *Masquerade
+	masquerades         chan *Masquerade
+	maxAllowedCachedAge time.Duration
+	maxCacheSize        int
+	cacheSaveInterval   time.Duration
+	cacheFile           string
```

Comment:
I would actually be tempted to remove this `cacheFile` as a variable as well -- i.e. just pass it to `initCaching` and don't store it @oxtoacart -- just again limiting the state if we don't need it.